alot/ui.py

```diff
@@ -664,3 +672,37 @@ def handle_signal(self, signum, frame):
             if isinstance(self.current_buffer, SearchBuffer):
                 self.current_buffer.rebuild()
                 self.update()
+
+    def cleanup(self):
+        """Do the final clean up before shutting down."""
+        self._save_history_to_file(self.commandprompthistory, self._cache)
+
+    @staticmethod
+    def _load_history_from_file(path):
+        """Load a history list from a file and split it into lines.
+
+        :param path: the path to the file that should be loaded
+        :type path: str
+        :returns: a list of history items (the lines of the file)
+        :rtype: list(str)
+
+        """
+        if os.path.exists(path):
+            return [line.rstrip('\n') for line in open(path).readlines()]
```

Comment:
This will return None if os.path.exists is False, but the documentation says it will always return a list of `str`s. Also, this use of open is not correct, it'll leak a file descriptor until gc runs.

I think that this function should be:

```python
if os.path.exists(path):
    with open(path, 'r') as f:
        return [l.rstrip('\n') for l in f]
return []
```

Note that iterating a file object creates iterator (non-materialized) instance rather than creating a concrete list.